cmd/snap-confine/snap-confine.c

```diff
@@ -574,8 +575,11 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 
 	/** Populate and join the device control group. */
 	struct snappy_udev udev_s;
-	if (snappy_udev_init(inv->security_tag, &udev_s) == 0)
-		setup_devices_cgroup(inv->security_tag, &udev_s);
+	if (snappy_udev_init(inv->security_tag, &udev_s) == 0) {
+		if (sc_cgroup_is_v2()) {
```

Comment:
if we are disabling cgroups when on cgroupv2, shouldn't this be 

```suggestion
		if (!sc_cgroup_is_v2()) {
```